pkg/sdk/renderer/argo/helpers.go

```diff
@@ -89,22 +89,35 @@ func getArgoArtifactRef(ref string) (*argoArtifactRef, error) {
 	ref = strings.TrimSuffix(ref, "}}")
 	parts := strings.Split(ref, ".")
 
+	invalidPathErrForRef := func(ref string) error {
+		return fmt.Errorf("invalid artifact path '%s'", ref)
```

Comment:
I think it should be more precise. Like "to many elements", "not enough elements"